### PR TITLE
add elfeed layer

### PR DESCRIPTION
**Pull request by d12frosted**
Resurrecting #1675

### CODE REVIEW

File: layers/+tools/elfeed/config.el [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+
+;;; Variables
+
+(defvar elfeed-web-enabled-on-emacs-startup nil
+  "If true, serve web interface Elfeed with simpl-httpd.")
+
+(defvar rmh-elfeed-org-files nil
```
Comment (nixmaniack) [inline layers/+tools/elfeed/config.el]: Does `rmh` have any significance, if not something more meaningful related to spacemacs, or just elfeed-org-files?

Comment (d12frosted) [inline layers/+tools/elfeed/config.el]: Ah. Good catch. Thanks. It's from `elfeed-org` - I'll remove it from `config.el` once I am home.

**Comment by d12frosted** [comment]
@nixmaniack done

**Comment by nixmaniack** [comment]
:+1:

**Comment by TheBB** [comment]
Thanks! Some review :-)

Can you fix up the readme according to the conventions? (No empty line after headlines, in particular.) Also please make install, usage and troubleshooting first-level headings. It doesn't look good when exporting if there is only one top level heading. Also please tag the ToC as `:TOC_4_org:noexport:` as is done in the updated README.org template.

`elfeed-org` and `elfeed-web` aren't deferred. Is this intentional? Should they be?

The improved `evilify` is called `spacemacs|evilify-map`. You should probably use that.

**Comment by d12frosted** [comment]
Thanks for review. Will ping you once it's done :smile:

**Comment by d12frosted** [comment]
@TheBB done. Thanks for pointing out that `elfeed-org` and `elfeed-web` are not deferred - it was a mistake.

**Comment by TheBB** [comment]
Thanks, but with the current setup I still think they are not deferred, since you are running code that will load the packages in the `:init` phase. We really should find a way to make them properly deferred because `elfeed-org` will for example force the loading of `org`, which is quite large.

**Comment by TheBB** [comment]
I suspect you can put `(elfeed-org)` into a use-package `:pre-config` hook for `elfeed` at least.

**Comment by d12frosted** [comment]
@TheBB updated, now `org` is not forced when you use `elfeed-org`. But when you set `elfeed-web-enabled-on-emacs-startup` to `t`, `elfeed` will be forced to be loaded as well as `org` (when you set `rmh-elfeed-org-files`), which makes sense I think.

**Comment by nixmaniack** [comment]
@d12frosted Since `rmh-elfeed-rog-files` has [default](https://github.com/remyhonig/elfeed-org/blob/master/elfeed-org.el#L52) value set will it force elfeed to load and as a dependency org to be loaded. Am I right? If that's the case, then we should start by setting that to `nil` and have user customize it's value. Also I don't like the default which would clutter `~/.emacs.d` on first startup.

**Comment by d12frosted** [comment]
@nixmaniack from what I see, `rmh-elfeed-rog-files` is set only after `elfeed-org` is called (that `defcustom` is not autoloaded). Elfeed layer calls `elfeed-org` only when `rmh-elfeed-rog-files` is set. So actually, it's safe to leave it as is. As for that file - it's not created for you if you don't have it (`elfeed-org` just errors when file doesn't exist). So from this point, it's also safe to leave this variable.

----

The only thing that is still not finished - mode map evilification - I am seeing warnings on `G`.

**Comment by nixmaniack** [comment]
@d12frosted Ah alright then. :smile: 
As for `G`, that's completely fine as it automatically remaps according to convention of auto-evilification. [This](https://github.com/syl20bnr/spacemacs/issues/2533) might be relevant if you want to understand reasoning behind it.

In this case just add bindings for `g` refresh view and `G` fetch feeds to something else. As a suggestion I might go for `r` for refresh and `l` for fetch(load) feeds (didn't suggest `f` as it had issues with evil-snipe etc. in past with magit) 

Hope it helps.

**Comment by d12frosted** [comment]
I see. Yeah, makes sense. 

Also, should I force `evil-state` when `vi` editing style users running `elfeed`? Cause right now they are in `normal` state by default.

**Comment by nixmaniack** [comment]
Evilified buffers should start in normal state. I think that's the default state if you use evilify macro.

Even if you use `vim`/`hybrid` editing style, all buffers by default start in `normal` state.

**Comment by d12frosted** [comment]
@nixmaniack thanks for the help - evilified state now works as expected.

**Comment by TheBB** [comment]
Evilified buffers should start in *evilified* state. That's what evilified state *is*. :-)

**Comment by d12frosted** [comment]
Yeah, the problem was because I haven't added `:mode elfeed-search-mode` (etc). But now everything is fine. Do you have any other concerns about elfeed layer?

**Comment by tsenart** [comment]
+1! Want this :)

**Comment by declanqian** [comment]
In the README.org file
(elfeed :variables
   rmh-elfeed-org-files ("~/.emacs.d/private/elfeed.org"))
should be:
(elfeed :variables
   rmh-elfeed-org-files (list "~/.emacs.d/private/elfeed.org"))

**Comment by syl20bnr** [comment]
Thank you ! :purple_heart: 
I made various fixes in commit https://github.com/syl20bnr/spacemacs/commit/2c9e68a and added the brand new `elfeed-goodies` (not yet in MELPA) in commit https://github.com/syl20bnr/spacemacs/commit/2cfd284
Cherry-picked into develop branch, you can safely delete your branch.

**Comment by darkfeline** [comment]
Thanks, but the fact that the `elfeed-org` package prefixes its member with `rmh-elfeed-org` really bugs me.  That's completely against well-established naming standards and ruins Spacemacs's Consistency.

**Comment by d12frosted** [comment]
thank you, @syl20bnr! `emacs-list` -> `emacs-lisp`. Haha.

**Comment by nixmaniack** [comment]
@darkfeline I very much agree! I did have comment on that part while reviewing but since the variable was defined upstream we used it as it is. It would be nice to change and make it consistent with Spacemacs.

**Pull request closed without merging**